pkg/action/dependency_test.go

```diff
@@ -159,3 +165,80 @@ func TestStatArchiveForStatus(t *testing.T) {
 	}
 	is.Equal("ok", statArchiveForStatus(where, dep))
 }
+
+func TestDependencyBuild_SkipRefresh(t *testing.T) {
+	client := NewDependency()
+	client.Verify = false
+
+	if v, ok := os.LookupEnv("GNUPGHOME"); ok {
+		client.Keyring = filepath.Join(v, "pubring.gpg")
+
+	} else {
+		client.Keyring = filepath.Join(homedir.HomeDir(), ".gnupg", "pubring.gpg")
+
+	}
+	srv, err := repotest.NewTempServerWithCleanup(t, "testdata/charts/*.tgz*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv.Stop()
+	if err := srv.LinkIndices(); err != nil {
+		t.Fatal(err)
+	}
+	rootDir := srv.Root()
+	baseURL := srv.URL()
+	chartname := "depbuildwithskiprefresh"
+
+	d := &chart.Chart{
+		Metadata: &chart.Metadata{
+			Name:       chartname,
+			Version:    "0.1.0",
+			APIVersion: "v1",
+			Dependencies: []*chart.Dependency{
+				{Name: "compressedchart", Version: "0.1.0", Repository: baseURL},
+			},
+		},
+	}
+	if err := chartutil.SaveDir(d, rootDir); err != nil {
+		t.Fatal(err)
+	}
+
+	// Set-up a manager
+	b := &bytes.Buffer{}
+	g := getter.Providers{getter.Provider{
+		Schemes: []string{"http", "https"},
+		New:     getter.NewHTTPGetter,
+	}}
+	m := &downloader.Manager{
+		ChartPath:        filepath.Join(rootDir, chartname),
+		Out:              b,
+		Getters:          g,
+		SkipUpdate:       client.SkipRefresh,
+		RepositoryConfig: filepath.Join(rootDir, "repositories.yaml"),
+		RepositoryCache:  rootDir,
+	}
```

Comment:
The test doesn't need to test the download manager. It needs to test command being run. Instead of looking in `dependency_test.go` try looking at the example in [`dependency_build_test.go`](https://github.com/helm/helm/blob/9aa0fbee6c118dfb0a32fa7026b27eebd0e74fb6/cmd/helm/dependency_build_test.go#L30). This is also where the test should go.